MoltenVK/MoltenVK/GPUObjects/MVKDescriptor.mm

```diff
@@ -161,10 +161,6 @@
                 MVKBuffer* buffer = (MVKBuffer*)bufferInfo.buffer;
                 bb.mtlBuffer = buffer->getMTLBuffer();
                 bb.offset = buffer->getMTLBufferOffset() + bufferInfo.offset;
-                if (bufferInfo.range == VK_WHOLE_SIZE)
-                    bb.size = (uint32_t)(buffer->getByteCount() - bb.offset);
-                else
-                    bb.size = (uint32_t)bufferInfo.range;
```

Comment:
Wait, isn't this used if the shader uses `OpArrayLength`? Admittedly, this is non-obvious from the surrounding code. You'd have to go to `Commands/MVKCommandEncoderState.mm` to see that.

Reply:
Ah. Thanks for saving me from myself. I thought I'd traced it through...so good catch! Reverted now.